### ISSUE TITLE
protocol v2: Type not in json

### ISSUE THREAD

**Issue by progrium** [ClusterHQ/powerstrip#41]
I think we agreed it would be good to have protocol version in headers since if we change the structure or content type, we'd not be able to check the version... and for similar but different reasons, I think `Type` should not be in JSON either. 

When you're used to Python/Ruby you don't think about this as much, but it really sucks to have data structures define what they inside themselves. Imagine Content-Type being part of the body. What!? How would you handle that? 

Using Go makes this obvious since we have to parse the payload twice to determine which type of struct to use when unmarshaling: https://github.com/ClusterHQ/powerstrip/blob/golang/protocol_test.go#L15

**Comment by lukemarsden** [comment]
I totally get this. But let's discuss whether we want to keep the JSON format or not in https://github.com/ClusterHQ/powerstrip/issues/44 before we decide to start moving things into HTTP headers.

**Comment by robhaswell** [comment]
@lukemarsden I disagree, I think we can continue to consider this without making a decision on #44.

I'm in favour of this proposal. We should be able to determine the schema of the body just from the header, and this includes both the version AND whether or not is is a pre/post hook.

:+1:
